### PR TITLE
Integrations v2.1: compatibility with hubs, devices and services

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Compatibility with core adjustment for integrations v2.1: hubs, devices & services.

Core PR: https://github.com/home-assistant/core/pull/80524

The frontend can do more with this integration-type information, but this PR mainly focuses on adding compatibility and keeping everything working as is.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/config_flow.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export const deleteConfigFlow = (hass: HomeAssistant, flowId: string) =>
 
 export const getConfigFlowHandlers = (
   hass: HomeAssistant,
-  type?: "helper" | "integration"
+  type?: Array<"device" | "helper" | "service">
```
Comment (bramkragten) [inline src/data/config_flow.ts:71]: why not `hub` here?

File: src/data/config_entries.ts [hunk 2]
```diff
@@ -54,7 +54,10 @@ export interface ConfigEntryUpdate {
 export const subscribeConfigEntries = (
   hass: HomeAssistant,
   callbackFunction: (message: ConfigEntryUpdate[]) => void,
-  filters?: { type?: "helper" | "integration"; domain?: string }
+  filters?: {
+    type?: Array<"device" | "helper" | "hub" | "service">;
```
Comment (bramkragten) [inline src/data/config_entries.ts:58]: We can extract this type?
```ts
type integrationType = "device" | "helper" | "hub" | "service";

Array<integrationType>;
```

File: src/data/config_flow.ts (hunk 1 above)
Comment (frenck) [reply src/data/config_flow.ts:71]: Because I'm an idiot? 😬 😄 

Thanks for spotting that 👍

**Comment by frenck** [comment]
The backend change has been approved. Once this PR is done, they both can be merged (they should be merged at the same time).

**Review by bramkragten — APPROVED** [review]

**Pull request merged**